10 Спринт

Melifaro2021 opened the pull request.

MaksBujanov committed:
10 Спринт.добавил:
1. SearchViewHolder.kt view песни
2. activity_song.xml разметка
3. Track.kt дата класс

MaksBujanov committed:
10 Спринт.добавил:
1. Glide.with в SearchViewHolder.kt
2. SearchAdapter.kt
3. RecyclerView
4. object MockPlaylist

MaksBujanov committed:
10 Спринт.добавил:
1. Ночную тему
2. Вынес размеры, строки

AleksandrIlinskii commented on app/build.gradle:
```diff
@@ -33,11 +33,21 @@ android {
+    //implementation 'com.google.android.material:material:1.9.0'
+
     implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
+    //implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
```
Неиспользуемые библиотеки нужно удалить, а не комментировать

Melifaro2021 replied:
Исправил.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/MediatekaActivity.kt:
```diff
@@ -2,12 +2,20 @@ package ru.makskim.playlist_maker
+
+        /*val image = findViewById<ImageView>(R.id.image)
+
+        Glide.with(applicationContext).load(imageUrl).into(image)*/
```
Все, что за комментировано - нужно удалить, оно тебе не пригодится

Melifaro2021 replied:
Исправил.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -12,20 +12,27 @@ import android.content.Context
 class SearchActivity : AppCompatActivity() {
     private lateinit var inputEditText: EditText // 1. Поле поиска
     private lateinit var searchQuery: String // 2. Поле ввода текста
     private lateinit var searchField: ConstraintLayout // 3. Иконка крестика
```
Иконка должна быть не ConstraintLayout, а обычный ImageView

Melifaro2021 replied:
Тут ошибка подписи. Это Поле поиска , а не иконка крестика. В поле поиска я могу использовать ConstraintLayout. Данная конструкция будет объяснена нам в 12 Спринте, но я ее уже знаю т.к. ранее учился в 8 когорте и мне были доступны эти лекции и я знаю как пользоваться ConstraintLayout.

Melifaro2021 replied:
Иконка крестика находится на строчке 31
        val clearButton = findViewById<ImageView>(R.id.clearSearchFieldIcon) // 3. Иконка крестика

Melifaro2021 replied:
Исправил.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -12,20 +12,27 @@ import android.content.Context
 
         searchField = findViewById(R.id.searchField) // 1. Поле поиска
         inputEditText = findViewById(R.id.inputEditText) // 2. Поле ввода текста
         val clearButton = findViewById<ImageView>(R.id.clearSearchFieldIcon) // 3. Иконка крестика
```
Эта строчка должна быть вместо то, что на 21

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -87,8 +94,10 @@ class SearchActivity : AppCompatActivity() {
+
     companion object {
         const val KEY = "search"
         const val DEF_SEARCH = ""
```
Для чего этот ключ?

Melifaro2021 replied:
Реализовывал пример из лекций.
Спринт 9/25 → Тема 4/5: Хранение данных. Начало → Урок 3/4 Хранение на практике

Пригодится ли ключ в дальнейшем в приложении не знаю.
![Пример](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/7329111e-0c69-479e-8811-e4ee81b27e06)

AleksandrIlinskii replied:
У тебя сейчас в companion object лежит 2 ключа. Хотя должен лежать 1 - для сохранения значения из поля поиска

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchViewHolder.kt:
```diff
@@ -0,0 +1,28 @@
+package ru.makskim.playlist_maker
```
Названия пакетов пишутся с верблюжьей раскладкой
https://kotlinlang.org/docs/coding-conventions.html#naming-rules

Melifaro2021 replied:
Изменил.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchViewHolder.kt:
```diff
@@ -0,0 +1,28 @@
+        // Ссылка на изображение обложки
+        Glide.with(itemView.context)
+            .load(item.artworkUrl100)
+            .transform(RoundedCorners(8))
```
Закругление 2, а не 8
<img width="270" alt="image" src="https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/19562b51-d853-4f2e-a7d5-837e0a3c1dba">

Melifaro2021 replied:
Исправил

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/Track.kt:
```diff
@@ -0,0 +1,7 @@
+package ru.makskim.playlist_maker
```
Не хватает пустой строки

Melifaro2021 replied:
Исправил. Объясните зачем нужна пустая строка?

avanisimov replied:
так принято

AleksandrIlinskii commented on app/src/main/res/layout/activity_mediateka.xml:
```diff
@@ -6,4 +6,18 @@
+            android:id="@+id/image"
+            android:layout_width="300dp"
+            android:layout_height="300dp"/>
+    </LinearLayout-->
```
Все комментарии надо удалить

Melifaro2021 replied:
Исправил.

MaksBujanov committed:
10 Спринт (промежуточные правки)

MaksBujanov committed:
10 Спринт (промежуточные правки2)

MaksBujanov committed:
10 Спринт (промежуточные правки3)
исправил строки
shareIntent.data = Uri.parse("mailto:")
shareIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(R.string.email_name))

MaksBujanov committed:
10 Спринт (промежуточные правки4)

MaksBujanov committed:
10 Спринт (промежуточные правки5)
Исправил критическую ошибку с переворотом экрана.

MaksBujanov committed:
10 Спринт (промежуточные правки6)
Проверка на наличие приложения для открытия пользовательского соглашения.

MaksBujanov committed:
10 Спринт (промежуточные правки7)
Изменил makskim.playlist_maker на makskim.playlistMaker

MaksBujanov committed:
10 Спринт (промежуточные правки8)
Цвет переключателя темной темы

Melifaro2021 commented:
Большая просьба укажите пожалуйста на каких именно страницах есть несоответствия по дизайну и в какой именно теме , если еще остались несоответствия. Заранее спасибо.

AleksandrIlinskii commented:
В прошлый раз я писал что плейсхолдер иконки не по макету. Сейчас он не исправлен 
Слева макет, справа приложение 
Чтобы получить плейсхолдер вместо иконок - удали приложение с телефона, выключи интернет и установи заново
<img width="592" alt="image" src="https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/eadf44c7-12ff-4dde-9bc8-ccc4ef786e33">

AleksandrIlinskii commented:
Прямо сейчас с репозитория не собирается приложение, видно ты поменял цвет свитча в настройках, но забыл запушить. Я подменил цвет для проверки
Старайся пушить все, что позволило бы собрать приложение сразу

Melifaro2021 commented:
> Прямо сейчас с репозитория не собирается приложение, видно ты поменял цвет свитча в настройках, но забыл запушить. Я подменил цвет для проверки Старайся пушить все, что позволило бы собрать приложение сразу

Запушить я не забыл. Ошибка была в другом,  в 92 строке в файле page_settings_style.xml была  не убрана строка дубликат для android:thumbTint , ее там быть не должно было, все уже было добавлено в activity_settings.xml. Так что просто убрал лишнюю строку.

Melifaro2021 commented:
> В прошлый раз я писал что плейсхолдер иконки не по макету. Сейчас он не исправлен Слева макет, справа приложение Чтобы получить плейсхолдер вместо иконок - удали приложение с телефона, выключи интернет и установи заново <img alt="image" width="592" src="https://user-images.githubusercontent.com/129396632/284690674-eadf44c7-12ff-4dde-9bc8-ccc4ef786e33.png">

Placeholder был прописан в файле SearchViewHolder.kt . Отображению правильной картинки по умолчанию мешала строка 27 видно на скриншоте. Я стер строку, теперь показывается правильное изображение (было установлено и ранее)
![image](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/45d59921-e79a-4a8d-a665-7d1eb394fc09)

Melifaro2021 commented:
> В прошлый раз я писал что плейсхолдер иконки не по макету. Сейчас он не исправлен Слева макет, справа приложение Чтобы получить плейсхолдер вместо иконок - удали приложение с телефона, выключи интернет и установи заново <img alt="image" width="592" src="https://user-images.githubusercontent.com/129396632/284690674-eadf44c7-12ff-4dde-9bc8-ccc4ef786e33.png">

Удалил приложение с телефона, выключил интернет , установил заново приложение, заскриншотил страницу поиска. Успел сделать скриншот до того как картинки подгрузились и после
![photo1700649788](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/24214999-c7f7-442d-b456-9c5ce2cad5ee)
![photo1700649816](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/a02dcb8d-661c-4e30-8c3d-957152b03882)

MaksBujanov committed:
10 Спринт (промежуточные правки8)
1. исправил android:thumbTint
2. исправил placeholder
3. сделал отступ от строки поиска в соответствии макетом

MaksBujanov committed:
10 Спринт (промежуточные правки9)
1. сделал отступ
